fem/fespace.cpp

```diff
@@ -990,10 +994,12 @@ FiniteElementSpace::RefinementOperator::RefinementOperator
    : fespace(fespace)
    , old_elem_dof(old_elem_dof)
 {
+#if 0 // with NC tet AMR this check no longer seems to work correctly in parallel
```

Comment:
This is strange: refining elements may not introduce new dofs in some cases but to get less dofs does not make sense.

Reply:
In rare cases this was happening in ex15p. Tets are bit strange and for sure the DOFs may get shuffled between processors when refining, so some processors may actually end up having fewer DOFs. But I agree that globally one should never get fewer DOFs, so I'll look a bit more into this.

Reply:
Any further insight into this?

Reply:
The check fails in time 0.86 of `mpirun -n 4 ex15p -m ../data/inline-tet.mesh`, I'm still looking why...

Reply:
I think I know what is going on. The first observation is that this run uses quadratic elements, so there are dofs on edges. The second observation is that the check counts both master and slave dofs (actually double counting shared dofs between processors). The reduced number of dofs (master + slave) then is due to a reduced number of edges (master + slave) in the finer mesh.

An example of this is a mesh that consists of one central tet with 4 face-adjacent tets where each of the 4 adjacent tets is refined uniformly once. This is the coarse mesh (with 33 = 1+4*8 tets). The fine mesh is obtained by refining the central tet. This refinement removes the 6 coarse edges of the central tet and introduces only one new edge -- the refinement diagonal of the inner octahedron. Thus the fine mesh has 5 edges less than the coarse.

Reply:
Thanks, I was close to finding this (I obtained two smaller meshes with this effect from the serial `ex15`), but you beat me to it :-) 

I'll change the check to use the true DOFs...

Reply:
Fixed. The checks are now based on the number of elements in the current and previous mesh.